src/main/java/org/broadinstitute/consent/http/enumeration/DraftType.java

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.consent.http.enumeration;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum DraftType {
+  @SerializedName("StudyDatasetSubmissionV1")
+  STUDY_DATASET_SUBMISSION_V1("StudyDatasetSubmissionV1");
```

Comment:
although there is only one value now, is the thought process here that having an enum for type makes the code more extensible/ future-proof? Or is this useful now too? Do we have plans for future draft types? Does this draft interface type have anything to do with the DraftStudyDataset class? If we have different classes implementing the Draft Interface type, do we need to store draft type on them or can their class indicate that information?

Reply:
Good questions!  This was part of a change Phil asked for to remove the reflective way I was doing that (by storing class information) and to use something the IDE and build tools would hint more at.
The DraftInterface (src/main/java/org/broadinstitute/consent/http/models/DraftInterface.java) now requires implementations to support a getDraftType (DraftType).  The DraftStudyDataset(src/main/java/org/broadinstitute/consent/http/models/DraftStudyDataset.java) declares the DraftType as static final and implements the getDraftType returning this value.

The enum value is persisted in the database.  You see it used in the src/main/java/org/broadinstitute/consent/http/db/mapper/DraftInterfaceMapper.java to instantiate the implementation.

As far as other draft types, consent already has them.  In the future, we can migrate them over to this more generalized approach.